### ISSUE TITLE
[Feature Request] Combine notes into one commit while publishing multiple notes

### ISSUE THREAD

**Issue by Mowd** [oleeskild/obsidian-digital-garden#115]
Publishing multiple notes creates a commit that only contains one note.
If I publish multiple, let's say, 5000+ notes at once, then that will be a long long time to wait.
I think combine multiple notes into one commit will significantly improve the speed.

<img width="694" alt="CleanShot_2022-12-02_02 12 56" src="https://user-images.githubusercontent.com/1097892/205128992-8f3ac6fb-3df3-4895-968c-cb79a1734738.png">

**Comment by ethanimproving** [comment]
Yes, this is super needed. I have a use case where I'm deploying to a Kubernetes cluster via Git Actions rather than Vercel or Netlify, and it is triggered by changes to the main brainch, so yesterday there were like 100 workflows all queued up at the same time. I definitely did not want that

![image](https://user-images.githubusercontent.com/55400451/228866092-e28f2290-edd5-4e89-83f1-581a76e79133.png)

**Comment by 1244919208** [comment]
Moreover, Vercel's free accounts have limited builds per day. Each time the Obsidian repo is re-pulled from the repository, digital-garden will think that every published file needs to be updated. This is very time consuming.

**Comment by Esurielt** [comment]
This is still super needed! Would be a game changer.

**Comment by jdhorner** [comment]
I have recently started using the plugin, and I couldn't agree more. I don't see any benefit to **not** grouping multiple changes into one commit. To have one per file is counterproductive when Vercel is involved.

**Comment by oleeskild** [comment]
I've finally gotten around to implement this. I've released it as a Beta version for now, that can be installed via BRAT. Any feedback, either here or in the [Discord](https://discord.gg/UsPH74nEVS) is much appreciated 🙏 

[Release](https://github.com/oleeskild/obsidian-digital-garden/releases/tag/2.57.0)
